server/routes/index.js

```diff
@@ -779,120 +765,84 @@ router
       .toArray()
     return res.status(200).json(data)
   })
-  .post(ensureUser, function (req, res) {
+  .post(ensureUser, async (req, res) => {
     const { appDb } = req.app.locals
-    if (Object.prototype.hasOwnProperty.call(req.body, 'disable')) {
-      appDb
-        .collection('configs')
-        .findOneAndUpdate(
-          { _id: new ObjectID(req.body.disable) },
-          { $pull: { readers: req.params.uid } },
-          { returnOriginal: false },
-          function (err) {
-            if (err) {
-              console.log(err)
-              return res.status(502).send({ message: 'fail' })
-            } else {
-              return res.status(201).send({ message: 'success' })
-            }
-          }
-        )
-    } else if (Object.prototype.hasOwnProperty.call(req.body, 'remove')) {
-      appDb
-        .collection('configs')
-        .deleteOne({ _id: new ObjectID(req.body.remove) }, function (err) {
-          if (err) {
-            console.log(err)
-            return res.status(502).send({ message: 'fail' })
-          } else {
-            return res.status(201).send({ message: 'success' })
-          }
+    try {
+      if (Object.prototype.hasOwnProperty.call(req.body, 'disable')) {
```

Comment:
I think we should create a ticket to refactor this function to make it more restful. IMO each of these `if/else` statements should be its own endpoint.

Reply:
<img width="1274" alt="Screen Shot 2022-11-14 at 5 07 15 PM" src="https://user-images.githubusercontent.com/19805355/201778210-1b76f310-9fea-46b2-a45f-0cdeee2eb0b9.png">